src/revisions_parser.py

```diff
@@ -2,37 +2,120 @@
 from typing import Dict, List
 
 
-def parse(input: str) -> List[Dict[str, str]]:
-    # find start
-    iter = re.finditer(r"revision history|version control", input,
-                       re.IGNORECASE)
-    start = None
-    a = 0
-    for start in iter:
-        a += 1
-        # first mention probably in toc
-        if a == 2:
-            break
-    if not start:
-        return []
-    input = input[start.span(0)[1]:]
+def month_to_number(input: str) -> str:
+    values = {
+        "jan": "01",
+        "feb": "02",
+        "mar": "03",
+        "apr": "04",
+        "may": "05",
+        "jun": "06",
+        "jul": "07",
+        "aug": "08",
+        "sep": "09",
+        "oct": "10",
+        "nov": "11",
+        "dec": "12",
+    }
+    form = input[:3].lower()
+    if form in values:
+        return values[form]
+    return ""
+
+
+def clean_date(input: str) -> str:
+    if "-" in input:
+        splitted = input.split("-")
+        if len(splitted) != 3:
+            return input
+
+        if not splitted[1].isnumeric():
+            # swap year and day
+            tmp = splitted[0]
+            splitted[0] = splitted[2]
+            splitted[2] = tmp
```

Comment:
The pythonic way of swapping is through assignment of pairs. Alternatively, `reversed` could be used here.